src/expression.c

```diff
@@ -11572,6 +11572,20 @@ Expression *AssignExp::semantic(Scope *sc)
                 if (e1x->checkPostblit(sc, t1))
                     return new ErrorExp();
             }
+
+            // e2 matches to t1 because of the implicit length match, so
+            if (isUnaArrayOp(e2x->op) || isBinArrayOp(e2x->op))
+            {
+                // convert e1 to e1[]
+                // e.g. e1[] = a[] + b[];
+                e1x = new SliceExp(e1x->loc, e1x, NULL, NULL);
+                e1x = e1x->semantic(sc);
```

Comment:
The bug is not only for `|` and `^`. If any array operations appear in e2 of static array assignment, it was broken. For example: `sarr = a[] + b[];` `sarr = a[] * b[];` `sarr = a[] / b[] - v;` ...